lib/Common/Memory/SmallFinalizableHeapBucket.h

```diff
@@ -167,39 +167,65 @@ class SmallHeapBlockType<FinalizableWithBarrierBit, MediumAllocationBlockAttribu
 template <class TBlockAttributes>
 class HeapBucketGroup
 {
-public:
     template <ObjectInfoBits objectAttributes>
-    typename SmallHeapBlockType<objectAttributes, TBlockAttributes>::BucketType& GetBucket()
+    class BucketGetter
     {
-        CompileAssert(objectAttributes & FinalizeBit);
-        return this->finalizableHeapBucket;
-    }
+    public:
+        static typename SmallHeapBlockType<objectAttributes, TBlockAttributes>::BucketType& GetBucket(HeapBucketGroup<TBlockAttributes> * heapBucketGroup)
```

Comment:
Could this be slightly more readable if you typedef BucketType here- can we use a typedef defined in the template class in the specialized class?